include/alpaka/kernel/TaskKernelOacc.hpp

```diff
@@ -0,0 +1,345 @@
+                                        threadElemExtent,
+                                        args...);
+                        },
+                        m_args));
+
+#if ALPAKA_DEBUG > ALPAKA_DEBUG_MINIMAL
+                std::cout << __func__
+                    << " blockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B" << std::endl;
+#endif
+                // The number of blocks in the grid.
+                TIdx const gridBlockCount(gridBlockExtent.prod());
+                // The number of threads in a block.
+                TIdx const blockThreadCount(blockThreadExtent.prod());
+
+                if(gridBlockCount == 0 || blockThreadCount == 0)
+                { //! empty grid is a NOP
+                    return;
+                }
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+                std::cout << "threadElemCount=" << threadElemExtent[0u]
+                    << "\tgridBlockCount=" << gridBlockCount << std::endl;
+#endif
+                auto argsD = m_args;
+                auto kernelFnObj = m_kernelFnObj;
+                dev.makeCurrent();
+                #pragma acc parallel num_workers(blockThreadCount) copyin(threadElemExtent,blockThreadExtent,argsD,gridBlockExtent) default(present)
+                {
+                    {
+                        #pragma acc loop gang
+                        for(TIdx b = 0u; b<gridBlockCount; ++b)
+                        {
+                            ctx::CtxBlockOacc<TDim, TIdx> blockShared(
+                                gridBlockExtent,
+                                blockThreadExtent,
+                                threadElemExtent,
+                                b,
+                                blockSharedMemDynSizeBytes);
+
+                            // Execute the threads in parallel.
+
+                            // Parallel execution of the threads in a block is required because when syncBlockThreads is called all of them have to be done with their work up to this line.
+                            // So we have to spawn one OS thread per thread in a block.
+                            //! \warning The OpenACC is technically allowed to ignore the value in the num_workers clause
+                            //! and could run fewer threads. The standard provides no way to check how many worker threads are running.
+                            //! If fewer threads are run, syncBlockThreads will dead-lock. It is up to the developer/user
+                            //! to choose a blockThreadCount which the runtime will respect.
+                            #pragma acc loop worker
+                            for(TIdx w = 0; w < blockThreadCount; ++w)
+                            {
+                                // blockThreadIdx[0] = w;
+                                ctx::CtxThreadOacc<TDim, TIdx> acc(
+                                    w,
+                                    blockShared);
```

Comment:
Is there a serious reason for these linebreaks?

Reply:
Style: Either all-in-one-line or one-per-line.

Reply:
Same format as used in the OpenMP backends in this place: https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp#L214 .